### PR TITLE
Introduce react-social-icons-lite to configure what icons to add to build (& Typescript types/build support)

### PR DESCRIPTION
**Pull request by pfernandom**
This PR introduces changes that will allow users of this package to considerably reduce their bundle size:

- Split the list of icons into individual files that can be imported individually.
- Created a script to split the icons file into individual icon files.
- Created a script to merge individual icons back into a single file (for backward compatibility)
- Introduced `react-social-icons-lite`, which is similar to `react-social-icons`, but it allows configuration only to load the icons that will be used.
- Introduced Typescript typings for the new component.
- Introduced Typescript configuration that allows better type support across the project, and Typescript builds.
- Introduced a new sample web app, that showcases the use of `react-social-icons-lite`, as well as tests for `react-social-icons-lite`.
- Updated the README.

The bundle size difference can be seen when building the sample projects:

```
npx parcel build examples/index.html
```

```
Built in 6.62s.

dist/app.650a7a15.js.map     356.48 KB     62ms
dist/app.650a7a15.js         225.82 KB    3.87s
dist/app.c13d0dca.css.map      2.76 KB      9ms
dist/app.c13d0dca.css          1.42 KB    1.17s
dist/index.html                 1011 B    2.40s
```

```
npx parcel build examples-lite/index.html
```
```
Built in 3.85s.

dist/app.6365a03e.js.map      270.8 KB    102ms
dist/app.6365a03e.js         130.83 KB    1.61s
dist/app.c13d0dca.css.map      2.77 KB      8ms
dist/app.c13d0dca.css          1.42 KB    928ms
dist/index.html                 1010 B    2.00s
```

The difference in bundle size is especially high if only a couple of icons are used. This would lead to a wider adoption of this package.

### Select icons to include in the bundle

A lightweight version of the component is provided in `react-social-icons/build/react-social-icons-lite.js`. This component requires individual icons to be loaded before use.

Each network icon can be imported from `react-social-icons/build/networks/`, like `react-social-icons/build/networks/facebook.js`.

```
import facebook from "react-social-icons/build/networks/facebook.js";
import github from "react-social-icons/build/networks/github.js";
import {
  SocialIcon,
  SocialIconDatabase,
} from "react-social-icons/build/react-social-icons-lite.js";

SocialIconDatabase
   .importNetwork(facebook)
   .importNetwork(github);

// ...

 <SocialIcon network="facebook" title="facebook" />
 <SocialIcon network="github" title="github" />
```

Importing only the used icons can reduce the final bundle size.

### CODE REVIEW

**Comment by couetilc** [comment]
Hi @pfernandom, thank you for your contribution. Code-splitting is an important feature and I appreciate your work on this. I’ve just released [v6 of react-social-icons](https://www.npmjs.com/package/react-social-icons) which implements code-splitting [in a different way](https://github.com/couetilc/react-social-icons/pull/166) ([see README.md](https://github.com/couetilc/react-social-icons)). I’d like to highlight a couple reasons why I decided to take a different approach.

1. Minimization of intermediary representations.

By removing choices, we are able to move faster. The more commands it takes to build the package, the more maintenance burden we take on as open-source developers. Embracing tools like rollup allow us to make seamless development experiences that take the manual work out of simple and useful APIs. Fewer intermediary representations means fewer files means less code means simpler to understand means simpler to contribute. Not relying on Typescript for such a simple package means fewer dependencies in the build chain easing the maintenance burden and widening the pool of developers that can contribute to this project.

2. As simple an API as possible with no breaking changes

My goal with v6 and code-splitting was to implement in such a way that there would be no breaking changes to 99.9% of use cases for this library. I was able to do that while making code-splitting a two-line change for anyone using this library (changing the default component export and adding an import statement for the specific icon you want to render). By embracing side effects and managing them through tree-shaking, we take advantage of simplifying and maintaining the interface users have with this package.

In conclusion.

Thank you for your work and I’m looking forward to your future contributions to react-social-icons.
